/**:
  ros__parameters:
    # Use dynamic map loading
    use_dynamic_map_loading: true

    # Vehicle reference frame
    base_frame: "base_link"

    # Subscriber queue size
    input_sensor_points_queue_size: 1

    # The maximum difference between two consecutive
    # transformations in order to consider convergence
    trans_epsilon: 0.01

    # The newton line search maximum step length
    step_size: 0.1

    # The ND voxel grid resolution
    resolution: 2.0

    # The number of iterations required to calculate alignment
    max_iterations: 30

    # Converged param type
    # 0=TRANSFORM_PROBABILITY, 1=NEAREST_VOXEL_TRANSFORMATION_LIKELIHOOD
    converged_param_type: 1

    # If converged_param_type is 0
    # Threshold for deciding whether to trust the estimation result
    converged_param_transform_probability: 3.0

    # If converged_param_type is 1
    # Threshold for deciding whether to trust the estimation result
    converged_param_nearest_voxel_transformation_likelihood: 2.3

    # The number of particles to estimate initial pose
    initial_estimate_particles_num: 100

    # Tolerance of timestamp difference between initial_pose and sensor pointcloud. [sec]
    initial_pose_timeout_sec: 1.0

    # Tolerance of distance difference between two initial poses used for linear interpolation. [m]
    initial_pose_distance_tolerance_m: 10.0

    # Number of threads used for parallel computing
    num_threads: 4

    # The covariance of output pose
    # Do note that this covariance matrix is empirically derived
    output_pose_covariance:
      [
        0.0225, 0.0,   0.0,   0.0,      0.0,      0.0,
        0.0,   0.0225, 0.0,   0.0,      0.0,      0.0,
        0.0,   0.0,   0.0225, 0.0,      0.0,      0.0,
        0.0,   0.0,   0.0,   0.000625, 0.0,      0.0,
        0.0,   0.0,   0.0,   0.0,      0.000625, 0.0,
        0.0,   0.0,   0.0,   0.0,      0.0,      0.000625,
      ]

    # Regularization switch
    regularization_enabled: false

    # regularization scale factor
    regularization_scale_factor: 0.01

    # Dynamic map loading distance
    dynamic_map_loading_update_distance: 20.0

    # Dynamic map loading loading radius
    dynamic_map_loading_map_radius: 150.0

    # Radius of input LiDAR range (used for diagnostics of dynamic map loading)
    lidar_radius: 100.0

    # cspell: ignore degrounded
    # A flag for using scan matching score based on de-grounded LiDAR scan
    estimate_scores_for_degrounded_scan: false

    # If lidar_point.z - base_link.z <= this threshold , the point will be removed
    z_margin_for_ground_removal: 0.8
